internal/serialization/compact_field.go

```diff
@@ -13,35 +13,205 @@ type CompactField struct {
 	Value any
 }
 
+func readNullableCompactField[T any](field string, f func(string) *T) any {
+	v := f(field)
+	if v == nil {
+		return nil
+	}
+	return *v
+}
+
 var compactReaders = map[serialization.FieldKind]compactFieldReader{
-	serialization.FieldKindString: func(r serialization.CompactReader, field string) any {
-		v := r.ReadString(field)
-		if v == nil {
-			return nil
-		}
-		return *v
+	serialization.FieldKindBoolean: func(r serialization.CompactReader, field string) any {
+		return r.ReadBoolean(field)
+	},
+	serialization.FieldKindArrayOfBoolean: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfBoolean(field)
+	},
+	serialization.FieldKindInt8: func(r serialization.CompactReader, field string) any {
+		return r.ReadInt8(field)
+	},
+	serialization.FieldKindArrayOfInt8: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfInt8(field)
+	},
+	// FieldKindChar        : Not decoded due to spec.
+	// FieldKindArrayOfChar : Not decoded due to spec.
+	serialization.FieldKindInt16: func(r serialization.CompactReader, field string) any {
+		return r.ReadInt16(field)
+	},
+	serialization.FieldKindArrayOfInt16: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfInt16(field)
 	},
 	serialization.FieldKindInt32: func(r serialization.CompactReader, field string) any {
 		return r.ReadInt32(field)
 	},
+	serialization.FieldKindArrayOfInt32: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfInt32(field)
+	},
+	serialization.FieldKindInt64: func(r serialization.CompactReader, field string) any {
+		return r.ReadInt64(field)
+	},
+	serialization.FieldKindArrayOfInt64: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfInt64(field)
+	},
+	serialization.FieldKindFloat32: func(r serialization.CompactReader, field string) any {
+		return r.ReadFloat32(field)
+	},
+	serialization.FieldKindArrayOfFloat32: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfFloat32(field)
+	},
+	serialization.FieldKindFloat64: func(r serialization.CompactReader, field string) any {
+		return r.ReadFloat64(field)
+	},
+	serialization.FieldKindArrayOfFloat64: func(r serialization.CompactReader, field string) any {
+		return r.ReadArrayOfFloat64(field)
+	},
+	serialization.FieldKindString: func(r serialization.CompactReader, field string) any {
```

Comment:
For FieldKindString we use "return readNullableCompactField(field, r.ReadString)". Is it correct?
( also for Decimal, Time, Date etc)

Reply:
Yes, those types are nullable.